### PR TITLE
http api and js client for dotnet interactive

### PR DESCRIPTION
**Pull request by colombod**
exposing command submission and event subscription via http api

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive.Js/tests/kernel-discovery-contract.spec.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import { expect } from "chai";
 import { createDotnetInteractiveClient } from "../src/dotnet-interactive/kernel-client-impl";
 import * as fetchMock from "fetch-mock";
-import { asKernelClientContainer, createMockKernelEventStream } from "./testSupprot";
+import { asKernelClientContainer, createMockKernelTransport } from "./testSupprot";
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Js/tests/kernel-discovery-contract.spec.ts:7]:
```suggestion
import { asKernelClientContainer, createMockKernelTransport } from "./testSupport";
```

File: src/Microsoft.DotNet.Interactive.Js/tests/testSupprot.ts [hunk 2]
```diff
@@ -12,16 +14,33 @@ export function configureFetchForKernelDiscovery(rootUrl: string) {
...
-                dispose: (): void => {
+export class MockKernelTransport implements KernelTransport {
+
+    public codeSubmissions: Array<KernelCommandEnvelope>;
+
+    constructor() {
+
+        this.codeSubmissions = new Array<KernelCommandEnvelope>();
+    }
+    public subscribeToKernelEvents(observer: KernelEventEvelopeObserver) {
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Js/tests/testSupprot.ts:25]:
```suggestion
    public subscribeToKernelEvents(observer: KernelEventEnvelopeObserver) {
```

File: src/dotnet-interactive/HttpRouting/KernelHub.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Reactive.Disposables;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.DotNet.Interactive.Events;
+using Microsoft.DotNet.Interactive.Server;
+
+namespace Microsoft.DotNet.Interactive.App.HttpRouting
```
Comment (jonsequitur) [inline src/dotnet-interactive/HttpRouting/KernelHub.cs:11]:
```suggestion
namespace Microsoft.DotNet.Interactive.App.Http
```

Comment (colombod) [reply src/dotnet-interactive/HttpRouting/KernelHub.cs:11]: Good point

**Pull request merged**